### PR TITLE
[RHPAM-3591] - Persist KIE Server maven and kie repositories

### PR DESCRIPTION
**Pull request by spolti**
Signed-off-by: spolti <fspolti@redhat.com>

See: https://issues.redhat.com/browse/RHPAM-3591

### CODE REVIEW

**Review by msmagnanijr — APPROVED** [review]
lgtm

**Review by desmax74 — APPROVED** [review]

**Comment by desmax74** [comment]
lgtm

File: pkg/controller/kieapp/defaults/defaults.go [hunk 1]
```diff
@@ -1701,6 +1716,14 @@ func isImmutable(cr *api.KieApp) bool {
 	return false
 }
 
+func isTrial(cr *api.KieApp) bool {
```
Comment (vaibhavjainwiz) [inline pkg/controller/kieapp/defaults/defaults.go:1719]: What is this trial mode?

**Review by vaibhavjainwiz — COMMENTED** [review]
Just one doubt. All looks good apart from it.

File: pkg/controller/kieapp/defaults/defaults.go (hunk 1 above)
Comment (spolti) [reply pkg/controller/kieapp/defaults/defaults.go:1719]: The Trial mode is the most simplier combination we can have when deploying RHPAM/DM.
It deploys the KIE Server and the Business/Decision Central with no PV and database used is embedded, the H2.
plus, it uses http routes.

Comment (vaibhavjainwiz) [reply pkg/controller/kieapp/defaults/defaults.go:1719]: thanks for explaining it.

**Pull request merged**